### ISSUE TITLE
[html] Auto-complete parentheses in <script> tags

### ISSUE THREAD

**Issue by ultraGentle** [microsoft/vscode#119336]
There is a very useful setting, `complete function calls`, which, when you accept a suggestion, completes both the parentheses and parameters of a function.  

Example:  you type `myFun` and accept the suggestion.  You get: `myFunction(param1, param2)`, where you can overtype param1 and param2.  

However: **Currently the setting only adds the `()` and function parameters in .js or .ts files, but not within `<script>` tags in a .html file.**   

It would be helpful if javascript language support in html files was as rich as in .js/.ts files.  

As with all FR's, if you are interested, please "thumbs up" this top post, since 20 votes are necessary for a successful FR.

**Comment by mjbvz** [comment]
1 problem per issue please. Let's use this to track support in script tags

I believe #119035 already tracks the second part

**Comment by ultraGentle** [comment]
Understood; edited original post to limit it to <script> tag support.

**Comment by ultraGentle** [comment]
@aeschli Would this be a good candidate for a PR?

**Comment by aeschli** [comment]
@ultraGentle A PR would be welcome! Let me know if you need guidance.

**Comment by ultraGentle** [comment]
Yes, I would appreciate guidance!  Maybe a pointer to the relevant spots, to start?  Or somewhere I could look for an example of how .js language features are integrated into .html docs?

**Comment by aeschli** [comment]
Code completion for JavaScript in HTML is done here: https://github.com/microsoft/vscode/blob/main/extensions/html-language-features/server/src/modes/javascriptMode.ts#L100

If you search for `useCodeSnippetsOnMethodSuggest` then you see the places where TypeScript does extra handling.
For TypeScript, the code that completes the function call is here: https://github.com/microsoft/vscode/blob/main/extensions/typescript-language-features/src/languageFeatures/completions.ts#L213

So basically what's done in TypeScript needs to be added to HTML as well.
The support in TypeScript is implemented against the TypeScript server API, while HTML embeds the TS library and works against the service API. They are quite similar but there might be surprises.

We have a few tests for JavaScript completion. You can find them here: https://github.com/microsoft/vscode/blob/main/extensions/html-language-features/server/src/test/completions.test.ts#L78

**Comment by ultraGentle** [comment]
Just wanted to make a note that I haven't had time to work on this, so if anyone does want to work on the PR, please feel free!
